flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -710,6 +711,9 @@ public long count() throws Exception {
 	 * @see DataSet
 	 */
 	public UnsortedGrouping<T> groupBy(String... fields) {
+		if (Keys.checkForAtomicType(getType(), fields)) {
```

Comment:
We need to check a few things here.
- If atomic type, we only allow 
  1. a single expression (`fields.length`)
  1. the expression must be a wildcard expression `"*"`
  1. the atomic type must be a key type (`TypeInformation.isKeyType() == true`)
- else do as before